File: backend/app/schemas/analytics.py
from typing import List

from pydantic import BaseModel


class TopicPerformance(BaseModel):
    topic: str
    total_attempts: int
    success_rate: float
    is_weakness: bool


class PerformanceReport(BaseModel):
    subject: str
    overall_score: float
    topics: List[TopicPerformance]
    weaknesses: List[str]
    strengths: List[str]


class PastQuestionIntelligence(BaseModel):
    subject: str
    frequently_repeated_topics: List[str]
    likely_exam_areas: List[str]
    revision_recommendations: List[str]

File: frontend/src/lib/api/analytics.ts
import { apiClient } from './client';

export interface TopicPerformance {
  topic: string;
  total_attempts: number;
  success_rate: number;
  is_weakness: boolean;
}

export interface PerformanceReport {
  subject: string;
  overall_score: number;
  topics: TopicPerformance[];
  weaknesses: string[];
  strengths: string[];
}

export interface PastQuestionIntelligence {
  subject: string;
  frequently_repeated_topics: string[];
  likely_exam_areas: string[];
  revision_recommendations: string[];
}

export const analyticsApi = {
  getPerformanceReport: async (subject: string) => {
    const { data } = await apiClient.get<PerformanceReport>(`/analytics/performance/${encodeURIComponent(subject)}`);
    return data;
  },

  getPastQuestionIntelligence: async (subject: string) => {
    const { data } = await apiClient.get<PastQuestionIntelligence>(`/analytics/past-questions/topics/${encodeURIComponent(subject)}`);
    return data;
  },
};
